client/src/store/restaurantReducer.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+
+import type { Restaurant } from '../flowTypes';
+import { Action } from 'redux';
+
+type State = {
+  restaurants: Array<Restaurant>,
+  isLoading: boolean
+};
+
+const initialState = { restaurants: [], isLoading: false };
+
+export const restaurantReducer = (state: State = initialState, action: Action): State => {
+  switch (action.type) {
+    case 'FETCH_RESTAURANTS_REQUEST':
+      return { ...state, isLoading: true };
```

Comment:
don't you want to set `restaurants` to empty array, to indicate a load is in progress?

Reply:
But the `restaurants` array might actually be empty isn't it 🤔

Reply:
The question is: when reloading the restaurants list - should there be any restaurants in the list, or should it reset until the server returns the full list?

Reply:
I changed the condition in `RestaurantList` so the loading spinner would appear if `isLoading && restaurants.length === 0`, so if we would like to allow the user to reload the list, his experience won't be ruined :)